src/main/java/org/folio/search/repository/IndexNameProvider.java

```diff
@@ -0,0 +1,40 @@
+package org.folio.search.repository;
+
+import static java.util.Locale.ROOT;
+import static org.folio.spring.tools.config.properties.FolioEnvironment.getFolioEnvName;
+
+import lombok.extern.log4j.Log4j2;
+import org.folio.search.domain.dto.ResourceEvent;
+import org.folio.search.model.ResourceRequest;
+import org.folio.search.model.index.SearchDocumentBody;
+import org.folio.search.service.consortium.TenantProvider;
+import org.springframework.stereotype.Component;
+
+@Log4j2
+@Component
+public class IndexNameProvider {
+
+  private final TenantProvider tenantProvider;
+
+  public IndexNameProvider(TenantProvider tenantProvider) {
+    this.tenantProvider = tenantProvider;
+  }
+
+  public String getIndexName(String resource, String tenantId) {
+    log.debug("Calculating index name for [resource: {}, tenantId: {}]", resource, tenantId);
```

Comment:
I guess it'd be good to move this one line down and print also calculated "finalTenantId"